pyproject.toml

```diff
@@ -86,6 +86,7 @@ all_extras = [
     "tslearn>=0.5.2; python_version < '3.11'",
     "xarray",
     "seasonal",
+    "scikit-optimize; python_version < '3.11'",
```

Comment:
Note: `numpy` version `1.24` does support `Python versions 3.8-3.11`. For some reason (pinning in other dependencies I guess), we are getting `1.24` only in `3.11`. When that reason is no longer applicable (i.e. when other dependencies upgrade their `numpy` pinning), how shall `sktime` be affected and what can we to resolve?

Reply:
I think that's an interesting but separate question. We have accumulated so many soft dependencies that they are interacting with each other in the `all_extras` set - how can we untangle this? As said, separate imo (unless we are introducing a worse bound here)

Reply:
Tracked in #4837